lightning-invoice/src/utils.rs

```diff
@@ -381,31 +399,43 @@ where
 /// channel with a higher or equal inbound capacity than `min_inbound_capacity_msat` exists
 /// * If any public channel exists, the returned `RouteHint`s will be empty, and the sender will
 /// need to find the path by looking at the public channels instead
-fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>) -> Vec<RouteHint>{
+fn filter_channels<L: Deref>(
+	channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>, logger: &L
+) -> Vec<RouteHint> where L::Target: Logger {
 	let mut filtered_channels: HashMap<PublicKey, &ChannelDetails> = HashMap::new();
 	let min_inbound_capacity = min_inbound_capacity_msat.unwrap_or(0);
 	let mut min_capacity_channel_exists = false;
 
+	log_debug!(logger, "Considering {} channels for invoice route hints", channels.len());
 	for channel in channels.iter() {
 		if channel.get_inbound_payment_scid().is_none() || channel.counterparty.forwarding_info.is_none() {
+			log_debug!(logger, "Ignoring channel {} for invoice route hints", log_bytes!(channel.channel_id));
 			continue;
 		}
 
 		if channel.is_public {
 			// If any public channel exists, return no hints and let the sender
 			// look at the public channels instead.
+			log_debug!(logger, "Not including invoice route hints on account of public channel {}",
```

Comment:
This wording will be pretty confusing for phantom invoices as well. Say a phantom invoice is generated using 3 participating nodes, where only one of the nodes have public channels, but the other 2 have only private channels. That invoice will include invoice route hints for the channels of the 2 participating nodes with private channels only, yet log this message due to the 3rd participating node with public channel(s), hence incorrectly indicating that no invoice route hints were included. 

Also note that phantom invoices will include invoice route hints for every participating node even if all of them have public channels. Phantom hops with the node id for every participating node will be included in the invoice route hints due to:
https://github.com/lightningdevkit/rust-lightning/blob/67726097c28f1ec5598ab9465e4de5a44225e56b/lightning-invoice/src/utils.rs#L178

Perhaps we could phrase this sentence as something like:
"Non of the considered channels are included in the invoice route hints on account of public channel {}."

Not sure what the best way to phrase this is though :)!